packages/babel-plugin-transform-es2015-duplicate-keys/src/index.js

```diff
@@ -14,13 +14,45 @@ export default function() {
         const { node } = path;
         const plainProps = node.properties.filter((prop) => !t.isSpreadProperty(prop) && !prop.computed);
 
-        const alreadySeenNames = Object.create(null);
+        // A property is a duplicate key if:
+        // * the property is a data property, and is preceeded by a data,
+        //   getter, or setter property of the same name.
+        // * the property is a getter property, and is preceeded by a data or
+        //   getter property of the same name.
+        // * the property is a setter property, and is preceeded by a data or
+        //   setter property of the same name.
+
+        const alreadySeenData = Object.create(null);
+        const alreadySeenGetters = Object.create(null);
+        const alreadySeenSetters = Object.create(null);
 
         for (let prop of plainProps) {
           const name = getName(prop.key);
-          if (!alreadySeenNames[name]) {
-            alreadySeenNames[name] = true;
-          } else {
+          let isDuplicate = false;
+          switch (prop.kind) {
+          case "get":
+            if (alreadySeenData[name] || alreadySeenGetters[name]) {
+              isDuplicate = true;
+            } else {
+              alreadySeenGetters[name] = true;
```

Comment:
Maybe for all of these you set `alreadySeenGetters[name] = true;` whether or not it's already seen (since it will be seen afterwards)